crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -628,10 +749,17 @@ mod tests {
     fn assert_public_ty(db: &TestDb, file_name: &str, symbol_name: &str, expected: &str) {
         let file = system_path_to_file(db, file_name).expect("Expected file to exist.");
 
-        let ty = public_symbol_ty_by_name(db, file, symbol_name).unwrap_or(Type::Unknown);
+        let ty = module_global_symbol_ty_by_name(db, file, symbol_name);
         assert_eq!(ty.display(db).to_string(), expected);
     }
 
+    impl TestDb {
+        fn write_dedented(&mut self, path: &str, content: &str) -> anyhow::Result<()> {
+            self.write_file(path, dedent(content))?;
+            Ok(())
+        }
+    }
+
```

Comment:
I rather have this block next to the `TestDb`. It's otherwise unclear where the `write_dedented` method is coming from. The implementation isn't local to this test block, it's available in the entire crate (but is private)